subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBundle.java

```diff
@@ -48,4 +52,18 @@ public DefaultSwiftBundle(String name, ProjectLayout projectLayout, ObjectFactor
     public DirectoryProperty getBundleDirectory() {
         return bundleDir;
     }
+
+    @Override
+    public FileCollection getCompileImportPath() {
+        return super.getCompileImportPath().plus(getFileOperations().files(new Callable<FileCollection>() {
+            @Override
+            public FileCollection call() throws Exception {
+                SwiftBinary tested = testedComponent.getOrNull();
+                if (tested == null) {
+                    return getFileOperations().files();
+                }
+                return getFileOperations().files(((DefaultSwiftBinary)tested).getObjectsDir());
+            }
```

Comment:
My only issue with this is `SwiftBundle` is not an XCTest specific bundle. We should discuss if `getCompileImportPath` and friend should return the `Configuration` so we can add dependency directly. I would need something similar for XCTest on Linux.